<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="en"><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8"/><link rel="stylesheet" href="../.resources/report.css" type="text/css"/><link rel="shortcut icon" href="../.resources/report.gif" type="image/gif"/><title>UserRepository.java</title><link rel="stylesheet" href="../.resources/prettify.css" type="text/css"/><script type="text/javascript" src="../.resources/prettify.js"></script></head><body onload="window['PR_TAB_WIDTH']=4;prettyPrint()"><div class="breadcrumb" id="breadcrumb"><span class="info"><a href="../.sessions.html" class="el_session">Sessions</a></span><a href="../index.html" class="el_report">car-pool</a> &gt; <a href="index.source.html" class="el_package">com.practo.carpool.repository</a> &gt; <span class="el_source">UserRepository.java</span></div><h1>UserRepository.java</h1><pre class="source lang-java linenums">package com.practo.carpool.repository;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.dao.DataAccessException;
import org.springframework.orm.hibernate5.HibernateTemplate;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

import com.practo.carpool.data.entity.User;
import com.practo.carpool.exceptions.NotFoundException;

@Repository
<span class="fc" id="L13">public class UserRepository implements CrudRepository&lt;User, Integer&gt; {</span>

  @Autowired
  protected HibernateTemplate hibTemp;

  @Override
  public Iterable&lt;User&gt; findAll() {
<span class="fc" id="L20">    return hibTemp.loadAll(User.class);</span>
  }

  @Override
  public User findOne(Integer id) throws NotFoundException {
    try {
<span class="fc" id="L26">      return hibTemp.load(User.class, id);</span>
<span class="nc" id="L27">    } catch (DataAccessException exception) {</span>
<span class="nc" id="L28">      throw new NotFoundException(&quot;No User found for the given Id&quot;);</span>
    }
  }

  @Override
  @Transactional
  public User save(User entity) throws NotFoundException {
    // try-catch will be triggered on attempting to update an entity which doesn't exist
    try {
<span class="fc" id="L37">      hibTemp.merge(entity);</span>
<span class="fc" id="L38">      return entity;</span>
<span class="nc" id="L39">    } catch (DataAccessException exception) {</span>
<span class="nc" id="L40">      throw new NotFoundException(&quot;No User found for the given Id&quot;);</span>
    }
  }
  

  @Override
  @Transactional
  public void delete(Integer id) throws NotFoundException {
<span class="nc" id="L48">    hibTemp.delete(findOne(id));</span>

<span class="nc" id="L50">  }</span>

}
</pre><div class="footer"><span class="right">Created with <a href="http://www.eclemma.org/jacoco">JaCoCo</a> 0.7.6.201602180812</span></div></body></html>